### ISSUE TITLE
3.0 - Fatal error when connecting PayPal and Banned Emails not empty.

### ISSUE THREAD

**Issue by mihaijoldis** [awesomemotive/easy-digital-downloads#9128]
### Expected behavior
Connecting PayPal (Commerce should work)

### Actual behavior
PayPal page (EDD) just shows the popups blocked message but got a log entry

### Steps to reproduce the behavior
1) Have at least 1 emails in the banned emails list.
2) Try to connect PayPal
3) Check the error log

### Information (if a specific version is affected):
PHP Version: 8.0.11

EDD Version (or branch): release/3.0

WordPress Version: latest

Any other relevant information:
Error log entry:
```
[11-Apr-2022 21:42:03 UTC] PHP Fatal error:  Uncaught TypeError: explode(): Argument #2 ($string) must be of type string, array given in D:\MAMP\htdocs\edd30\wp-content\plugins\easy-digital-downloads\includes\admin\settings\register-settings.php:1603
Stack trace:
#0 D:\MAMP\htdocs\edd30\wp-content\plugins\easy-digital-downloads\includes\admin\settings\register-settings.php(1603): explode('\n', Array)
#1 D:\MAMP\htdocs\edd30\wp-includes\class-wp-hook.php(307): edd_sanitize_banned_emails(Array, 'banned_emails')
#2 D:\MAMP\htdocs\edd30\wp-includes\plugin.php(189): WP_Hook->apply_filters(Array, Array)
#3 D:\MAMP\htdocs\edd30\wp-content\plugins\easy-digital-downloads\includes\admin\settings\register-settings.php(1264): apply_filters('edd_settings_sa...', Array, 'banned_emails')
#4 D:\MAMP\htdocs\edd30\wp-includes\class-wp-hook.php(309): edd_settings_sanitize(Array)
#5 D:\MAMP\htdocs\edd30\wp-includes\plugin.php(189): WP_Hook->apply_filters(Array, Array)
#6 D:\MAMP\htdocs\edd30\wp-includes\formatting.php(4976): apply_filters('sanitize_option...', Array, 'edd_settings', Array)
#7 D:\MAMP\htdocs\edd30\wp-includes\option.php(421): sanitize_option('edd_settings', Array)
#8 D:\MAMP\htdocs\edd30\wp-content\plugins\easy-digital-downloads\includes\admin\settings\register-settings.php(79): update_option('edd_settings', Array)
#9 D:\MAMP\htdocs\edd30\wp-content\plugins\easy-digital-downloads\includes\gateways\paypal\admin\connect.php(257): edd_update_option('paypal_sandbox_...', 'AYxz0bB_C53V_yA...')
#10 D:\MAMP\htdocs\edd30\wp-includes\class-wp-hook.php(307): EDD\Gateways\PayPal\Admin\get_and_save_credentials('')
#11 D:\MAMP\htdocs\edd30\wp-includes\class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
#12 D:\MAMP\htdocs\edd30\wp-includes\plugin.php(474): WP_Hook->do_action(Array)
#13 D:\MAMP\htdocs\edd30\wp-admin\admin-ajax.php(187): do_action('wp_ajax_edd_pay...')
#14 {main}
  thrown in D:\MAMP\htdocs\edd30\wp-content\plugins\easy-digital-downloads\includes\admin\settings\register-settings.php on line 1603
  ```

**Comment by mihaijoldis** [comment]
If I see it right the banned email sanitize function will run every time the admin settings are saved?
`add_filter( 'edd_settings_sanitize', 'edd_sanitize_banned_emails', 10, 2 );`

Wonder if that should be changed to run on a different filter like everything else?
`apply_filters( 'edd_settings_sanitize_' . $type, $output[ $key ], $key );`

**Comment by mihaijoldis** [comment]
Also maybe we can change that sanitize function for moderation to run just when it needs to using:
`add_filter( 'edd_settings_general-moderation_sanitize'`

**Comment by robincornett** [comment]
I think it make sense to use the tab/section filter hook instead of the general one. However, #9121 is moving the setting so we need to track with that PR.

It looks like the emails were not being sanitized at all and then that was fixed in #8513. This is broken--for me, PayPal Commerce does actually connect, but with the fatal error. Additionally, because the logic fails, the banned emails setting is actually erased, and that's not great.

When the filter is running during the ajax request, the `banned_emails` setting is already an array, and that's not accounted for in the `edd_sanitize_banned_emails` function. It looks like when PayPal Commerce is connected, all settings are sanitized because it's not considered a "section" by the `edd_settings_sanitize` logic, which is why the banned emails are being sanitized again.
